code/weapon/weapons.cpp

```diff
@@ -2173,6 +2174,11 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 	if( optional_string("$Leech Afterburner:") ){
 		stuff_float(&wip->afterburner_reduce);
 	}
+	
+	// Multiplier for the healing done to the shooter of the weapon.
+	if( optional_string("$Vampiric Healing Factor:") ){
+		stuff_float(&wip->succ_factor);
```

Comment:
Costing health to shoot is a cool idea, but I think should be handled separately, and therefore probably best as a separate field entirely. Given that, you should probably make sure that value is not negative. And perhaps also warn if the "vampiric" flag hasn't been used yet they are trying to set this factor? Or the reverse, they used the flag but failed to set this value.